api/routes/download.js

```diff
@@ -0,0 +1,88 @@
+var express = require('express');
+var aws_secret = require('../utils/s3_config.json');
+var AWS = require('aws-sdk');
+var router = express.Router();
+AWS.config.update(aws_secret);
+var s3 = new AWS.S3();
+
+// GET for profile image
+//@parms should send "userName" as a query parameter
+//@returns aws url
+router.get('/profile', function (req,res) {
+  console.log('got the request for profile');
+  
+  var urlParams = {Bucket: 'project-match', Key: 'profile/' + req.query.userName };
+  console.log(urlParams);
+    s3.getSignedUrl('getObject', urlParams, function(err, url){
+      if(err){
+        res.send({error: err.message});
+      }
+      console.log('the url of the image is', url);
+      res.send({url: url});
+  });
+})
+
+// GET for project images

```

Comment:
to clarify, we would upload images with the upload route. then we can call all images associated with a projectID with the download route. 

And wanted to confirm in front-end that I should save a new project first, and then once the _ID is returned, save any associated images that were uploaded. Then reconcile the imageLinks with the projectObject somehow?

Reply:
ya, that is how I designed the API. you can upload images by project id and because the image name have timestamps you can put them in the order